ledger/src/shred.rs

```diff
@@ -448,33 +442,68 @@ impl Shred {
         self.common_header.slot
     }
 
-    pub fn parent(&self) -> Option<Slot> {
+    pub fn parent(&self) -> Result<Slot> {
         match self.shred_type() {
             ShredType::Data => {
-                let parent_offset = Slot::try_from(self.data_header.parent_offset);
-                self.slot().checked_sub(parent_offset.ok()?)
+                // TODO What if parent_offset is zero and slot != 0?
+                let parent_offset = Slot::from(self.data_header.parent_offset);
+                self.slot().checked_sub(parent_offset).ok_or_else(|| {
+                    ShredError::InvalidParentOffset {
+                        slot: self.slot(),
+                        parent_offset: self.data_header.parent_offset,
+                    }
+                })
             }
-            ShredType::Code => None,
+            ShredType::Code => Err(ShredError::InvalidShredType),
         }
     }
 
     pub fn index(&self) -> u32 {
         self.common_header.index
     }
 
+    pub(crate) fn fec_set_index(&self) -> u32 {
+        self.common_header.fec_set_index
+    }
+
+    // Returns true if the shred sanity checks.
```

Comment:
nit: `if the shred passes sanity checks`

Reply:
done!